File: rss-reader/app/src/main/java/com/abubusoft/kripton/examples/rssreader/service/persistence/DaoBase.java
package com.abubusoft.kripton.examples.rssreader.service.persistence;

import com.abubusoft.kripton.android.annotation.BindSqlInsert;
import com.abubusoft.kripton.android.sqlite.ConflictAlgorithmType;

public interface DaoBase<E> {
    @BindSqlInsert(conflictAlgorithm = ConflictAlgorithmType.IGNORE)
    boolean insert(E bean);
}

File: kotlin-rss-reader/app/src/main/java/com/abubusoft/rssreader/service/persistence/DaoBase.kt
package com.abubusoft.rssreader.service.persistence

import com.abubusoft.kripton.android.annotation.BindSqlInsert
import com.abubusoft.kripton.android.sqlite.ConflictAlgorithmType

interface DaoBase<E> {
    @BindSqlInsert(conflictAlgorithm = ConflictAlgorithmType.IGNORE)
    fun insert(bean: E): Boolean
}